lib/smart_proxy_container_gateway/container_gateway_main.rb

```diff
@@ -40,6 +40,14 @@ def pulp_registry_request(uri, headers)
         end
       end
 
+      def flatpak_static_index(headers, params = {})
+        uri = URI.parse("#{@pulp_endpoint}/pulpcore_registry/index/static")
+        unless params.empty?
+          uri.query = URI.encode_www_form(params)
```

Comment:
For now, I'm going with `encode_www_form` since `encode_uri_component` isn't available in the versions of Ruby we use (3.0).

Reply:
I did some research before and `+` subbed for ` ` generally seemed okay in the query string, I suppose that is what the blog refers to in the "Why is this so confusing and weird?" section. It's clear anyway that `%20` is most correct. Including ERB for this does feel a little odd, but it sounds safe enough.

Reply:
Updated to use `ERB::Util.url_encode`